core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/WorkflowTools.java

```diff
@@ -0,0 +1,331 @@
+package org.silverpeas.core.workflow.engine;
+
+import org.silverpeas.core.contribution.content.form.Field;
+import org.silverpeas.core.util.logging.SilverLogger;
+import org.silverpeas.core.workflow.api.TaskManager;
+import org.silverpeas.core.workflow.api.UserManager;
+import org.silverpeas.core.workflow.api.WorkflowException;
+import org.silverpeas.core.workflow.api.event.GenericEvent;
+import org.silverpeas.core.workflow.api.instance.Actor;
+import org.silverpeas.core.workflow.api.instance.Question;
+import org.silverpeas.core.workflow.api.instance.UpdatableHistoryStep;
+import org.silverpeas.core.workflow.api.instance.UpdatableProcessInstance;
+import org.silverpeas.core.workflow.api.model.Action;
+import org.silverpeas.core.workflow.api.model.Consequence;
+import org.silverpeas.core.workflow.api.model.Consequences;
+import org.silverpeas.core.workflow.api.model.ProcessModel;
+import org.silverpeas.core.workflow.api.model.QualifiedUsers;
+import org.silverpeas.core.workflow.api.model.State;
+import org.silverpeas.core.workflow.api.model.Trigger;
+import org.silverpeas.core.workflow.api.task.Task;
+import org.silverpeas.core.workflow.api.user.User;
+import org.silverpeas.core.workflow.external.ExternalAction;
+
+import java.util.Hashtable;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * A TimeoutRequest indicates the workflow engine that an instance is in an active state since a
+ * too long period
+ */
+class WorkflowTools {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1118)